analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -97,7 +97,7 @@ public override void ExecutionCompleted()
     protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
         if (context.Operation.Instance.AsInvocation() is { } invocation
-            && invocation.Instance is { } instance
+            && invocation.Target(context.State) is { } instance
```

Comment:
This rule doesn't raise for extension methods (yet). This would be a next step (see #7457). Until then, this change is not needed.